packages/openapi-generator/src/cli/index.ts

```diff
@@ -62,27 +61,29 @@ $ generate-openapi-client --input ./my-spec.yaml --outputDir ./client --transpil
     strictNaming: flags.boolean({
       name: 'strictNaming',
       description:
-        'By default, the generation fails, when there are duplicate names for operations and/or path parameters after transforming them to camel case. Set this to true to enable unique name generation. The names will then be generated by appending numbers.',
+        'By default, the generation fails, when there are duplicate names for operations and/or path parameters after transforming them to camel case. Set this to true to enable unique name generation. The names are then be generated by appending numbers.',
       default: true,
-      required: false,
       allowNo: true
     }),
     tsConfig: flags.string({
       description:
-        'Replace the default `tsconfig.json` by passing a path to a custom config. By default, a `tsconfig.json` is only generated, when transpilation is enabled (`--transpile`).',
+        'Replace the default `tsconfig.json` by passing a path to a custom config. By default, a `tsconfig.json` is only generated, when transpilation is enabled (`--transpile`). If a directory is passed, a `tsconfig.json` file is read from this directory.',
       parse: input => resolve(input),
       helpValue: '<path/to/tsconfig.json>'
     }),
     packageJson: flags.boolean({
       description:
-        'When set to false, no `package.json` is generated. By default, a `package.json` that specifies dependencies and scripts for transpilation and documentation generation is generated.',
-      default: true,
-      allowNo: true
+        'When set to true, a `package.json`, that specifies dependencies and scripts for transpilation and documentation generation is generated.',
+      default: false
     }),
-    serviceMapping: flags.string({
+    perServiceConfig: flags.string({
       description:
-        'Set the path to the service mapping file. By default, a `service-mapping.json` is generated in the input directory. The service mapping ensures consistent names between multiple generation runs with updated service definitions.',
-      helpValue: '<path/to/custom-service-mapping.json>'
+        'Set the path to a file containing the configuration per service. The configuration allows to set a `directoryName` and `packageName` for every service. It also makes sure that names do not change between generator runs. If a directory is passed, a `per-service-config.json` file is read/created in this directory.',
+      helpValue: '<path/to/per-service-config.json>',
+      parse: input =>
+        extname(input)
```

Comment:
checking extension name might not be enough, because you might pass `config` without extension and is used as folder here.

Reply:
Good point, will fix.

Reply:
Actually, should we then always treat this as a file and refrain from adding a default name?

Reply:
Forget this question. Done.